src/Plugin/Commerce/ShippingMethod/CanadaPost.php

```diff
@@ -147,11 +147,11 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
     $form['api']['info'] = [
       '#type' => 'markup',
       '#markup' => $this->t(
-        'Please @action your @link.',
+        'Canada Post API settings can be added per store or can be configured to be used sitewide. Please @action your @link here.',
```

Comment:
**CHANGE REQUEST**

I think we got this wrong. We need shipping method + store entity level configuration, not sitewide. If it's easy, let's do shipping method as well, if not let's keep store entity only for the stable release. Permanently remove sitewide configuration, it's not going to be added back.

Reply:
Dimitris, I've now refactored everything so that we have settings per shipping method + per store settings. The store level settings get preference. so, the sitewide config is now permanently deleted.

I've also refactored the code so that we now have a general Request class that serves up the API settings and auth info which the Ratings/Tracking/Utilities classes extend.